ocrd/cli/workspace.py

```diff
@@ -0,0 +1,146 @@
+import os
+
+import click
+
+from ocrd import Resolver, WorkspaceValidator, Workspace
+
+class WorkspaceCtx(object):
+
+    def __init__(self, directory):
+        self.directory = directory
+        self.resolver = Resolver(cache_enabled=True)
+        self.config = {}
+        self.verbose = False
+
+pass_workspace = click.make_pass_decorator(WorkspaceCtx)
+
+# ----------------------------------------------------------------------
+# ocrd workspace
+# ----------------------------------------------------------------------
+
+@click.group("workspace", help="Working with workspace")
+@click.option(
+    '-d',
+    '--directory',
+    envvar='WORKSPACE_DIR',
+    default=os.path.abspath('.'),
+    type=click.Path(file_okay=False),
+    metavar='PATH',
+    help='Changes the repository folder location.'
+)
+@click.option(
+    '-c',
+    '--config',
+    nargs=2,
+    multiple=True,
+    metavar='KEY VALUE',
+    help='Overrides a config key/value pair.'
+)
+@click.option(
+    '-v',
+    '--verbose',
+    is_flag=True,
+    help='Enables verbose mode.'
+)
+@click.pass_context
+def workspace_cli(ctx, directory, config, verbose):
+    ctx.obj = WorkspaceCtx(os.path.abspath(directory))
+    ctx.obj.verbose = verbose
+    for key, value in config:
+        ctx.obj.config[key] = value
+
+# ----------------------------------------------------------------------
+# ocrd workspace validate
+# ----------------------------------------------------------------------
+
+@workspace_cli.command('validate', help='''
+
+    Validate a workspace
+
+''')
+@click.option('-m', '--mets-url', help="METS URL to validate")
+@pass_workspace
+def validate_workspace(ctx, mets_url=None):
+    if mets_url is None:
+        mets_url = 'file://%s/mets.xml' % ctx.directory
+    report = WorkspaceValidator.validate_url(ctx.resolver, mets_url)
+    print(report.to_xml())
+    if not report.is_valid:
+        return 128
```

Comment:
Magic number?

Reply:
Yeah, `128` is random. Must just be something non-zero, so you can do

```sh
if ocrd workspace validate -m https://server/path/to/mets.xml;then
  echo 'ok'
else
	echo 'not ok'
fi
```

But it didn't work because return of the click handler !== retval of process. I'll fix that.